bfg/src/test/scala/com/madgag/git/bfg/cli/MainSpec.scala

```diff
@@ -110,6 +110,12 @@ class MainSpec extends Specification {
         run("--delete-folders bar --delete-files foo.txt")
       }
     }
+    
+    "not crash on encountering a commit with bad encoding header" in new unpackedRepo("/sample-repos/badEncoding.git.zip") {
```

Comment:
The important commit object within this repo:

```
$ git cat-file -p 0554b30af831719d1de53ee2ee8ff18d3f637953
tree 2b297e643c551e76cfa1f93810c50811382f9117
author Brett Randall <javabrett@gmail.com> 1476143356 +1100
committer Brett Randall <javabrett@gmail.com> 1476143356 +1100
encoding utf-8,ISO-8859-1

This commit message will have a bad Encoding header of Encoding utf-8,ISO-8859-1
```

... the important line being `encoding utf-8,ISO-8859-1` - ie an invalid encoding.